source/includes/connection-snippets/x509-snippets/csharp-connection.cs

```diff
@@ -0,0 +1,51 @@
+// begin x509 connection
+using MongoDB.Bson;
+using MongoDB.Driver;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+
+// Tested using MongoDB.Driver 2.9.2 and netcoreapp 2.2
+
+namespace WorkingWithMongoDB
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            MainAsync().Wait();
+        }
+
+        static async Task MainAsync()
+        {
+            var connectionString = "mongodb+srv://<cluster-url>/test?authSource=$externalretryWrites=true&w=majority&authMechanism=MONGODB-X509";
```

Comment:
Is `authSource=$external` actually required?

There seems to be a separator missing between the `authSource` value and the `retryWrites` keyword.

Reply:
* `authSource=$external` is technically not needed for csharp, but it is needed for other drivers, and for these snippets, we would ideally be as consistent amongst drivers as possible
* Updated the uri to include the separator